File: Base/조합/조합.py
from pprint import pprint


def COMB(arr, r):
    result = []

    def comb(k, chosen, start):
        if k == r:
            result.append(chosen[::])
            return
        for i in range(start, len(arr)):
            chosen.append(arr[i])
            comb(k+1, chosen, i+1)
            chosen.pop()
    comb(0, [], 0)
    return result


result = COMB('ABCDE', 2)
pprint(result)
pprint(str(len(result)) + "개")

File: 자스베이스/순열조합/순열조합/조합.js
const comb = (k, N, start, choice, numbers) => {
    if (k === N) {
        console.log(choice.join(" "));
        return;
    }
    for (let i = start; i < numbers.length; i++) {
        choice.push(numbers[i]);
        comb(k + 1, N, i + 1, choice, numbers);
        choice.pop();
    }
};

comb(0, 2, 0, [], [1, 2, 3]);
